### DISCUSSION TITLE
Can you show declared schemas in ReDoc separately, like it's done by default in Swagger?

### DISCUSSION THREAD

**Discussion by mrestof-vre** [fastapi/fastapi#11165] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from pydantic import BaseModel

class Hero(BaseModel):
    name: str
    age: int

app = FastAPI()

@app.get('/')
def get_hero() -> Hero:
    return Hero(name='Superman', age=22)
```

### Description

When I open Swagger auto generated docs, I can see specs for endpoints and schemas separately. But when I open ReDoc auto generated docs I don't see separate schema definitions. All I found is [this article](https://redocly.com/blog/display-schemas/), that describes how to configure redocly cli tool to add separate schemas support. But I don't know if it's possible to enable the described behavior with fastapi auto generated docs. I would really appreciate any help, thanks in advance.

### Operating System

Linux

### FastAPI Version

0.109.2

### Pydantic Version

2.6.1

### Python Version

3.12.1

**Comment by hjoukl** [comment]
Hi, 

a workaround to approximate the desired behaviour is this use of the `openapi_tags` arg:

```
from fastapi import FastAPI
from pydantic import BaseModel, Field

# ...

class HealthOkResponse(BaseModel):
    # ...


description = """# API description

## Crucial informations on API usage
...
"""

app = FastAPI(
    title="Summary API",
    summary="API for summaries upload",
    description=description,
    openapi_tags=[
        {"name": "summaries",
         "description": "Operations for summaries", },
        {"name": "api-infos",
         "description": "Operations for API health and metadata", },
        {"name": "datamodels",
         "x-displayName": "Data Models",
         "description":
            'HealthOkResponse\n'
            '<SchemaDefinition schemaRef="#/components/schemas/HealthOkResponse"'
            ' showReadOnly={true} showWriteOnly={true}/>\n'
            'HealthErrorResponse\n'
            '<SchemaDefinition schemaRef="#/components/schemas/HealthErrorResponse"'
            ' showReadOnly={true} showWriteOnly={true}/>',
        },
        ],
    version="0.1.0",
)
```

The crucial part is the "description" entry in the `{"name": "datamodels", ...}` item.

This makes use of the `<SchemaDefinition .../>` html stuff mentioned in the article you linked, which redocly renders in the docs.

You'll need to repeat all the data model names you want to show up in the docs, so it's not very DRY and pretty explicit.
But it seems to work. :-)

hth,
Holger

**Comment by YuriiMotov** [comment]
Let's close this for now to keep the discussions clean and organized.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.

**Comment by nkrishnaswami** [comment]
BTW, if we tweak `get_redoc_html` to include `schema-definitions-tag-name`, e.g.,:

```
    <redoc spec-url="{openapi_url}" schema-definitions-tag-name="Schema definitions"></redoc>
```

the schemas will be listed alphabetically under that tag, eg "Schema definitions".  I haven't gotten examples set via `json_schema_extra` to show up here, however (they do under the API bodies)
